### PR TITLE
[IOTDB-6182] Add fsync in ConfigNode persistence modules

### PR DESCRIPTION
**Pull request by CRZbulabula**
Use fsync when the ConfigNodes need to take snapshot, which is safer than previous implementation.

### CODE REVIEW

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/PartitionInfo.java [hunk 1]
```diff
@@ -849,34 +850,37 @@ public boolean processTakeSnapshot(File snapshotDir) throws TException, IOExcept
...
       for (RegionMaintainTask task : regionMaintainTaskList) {
-        task.serialize(fileOutputStream, protocol);
+        task.serialize(bufferedOutputStream, protocol);
       }
 
       // write to file
-      fileOutputStream.flush();
-      fileOutputStream.close();
+      tioStreamTransport.flush();
+      bufferedOutputStream.flush();
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/PartitionInfo.java:878]: It looks like this line is redundant.

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/PartitionInfo.java [hunk 2]
```diff
@@ -849,34 +850,37 @@ public boolean processTakeSnapshot(File snapshotDir) throws TException, IOExcept
...
+      fileOutputStream.getFD().sync();
+
       // rename file
       return tmpFile.renameTo(snapshotFile);
+
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/PartitionInfo.java:883]: remove this empty line

**Review by OneSizeFitsQuorum — COMMENTED** [review]
PTAL

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/PartitionInfo.java (hunk 1 above)
Comment (CRZbulabula) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/PartitionInfo.java]: Removed

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/PartitionInfo.java (hunk 2 above)
Comment (CRZbulabula) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/PartitionInfo.java]: Removed

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM~

**Review by wangchao316 — APPROVED** [review]
LGTM

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM~

**Pull request merged**